### ISSUE TITLE
How to call Stored Procedure from a DBPool ?

### ISSUE THREAD

**Issue by Didier68** [IBM/nodejs-idb-pconnector#148]
Hello, I'm trying to call stored procedures and functions either directly or via an SQL query, but it consistently fails with idb-pconnector (even though I successfully performed the same test with **idb-connector** and **Mapepire** without any issues...).

In the example below, I'm calling a function "F_PRIMA" (in RPGLE) from an SQL query. This function is located in the ERPZPRG library, which is also the "CURLIB" and is the first library from my LIBLIST.

		const idbPConnector = require('idb-pconnector');
		const { dbconn, SQL_ATTR_DBC_SYS_NAMING, SQL_TRUE } = require('idb-connector');

		async function testsql() {
			
			console.log("Start...");
			let pool = null;
			let poolConnection =  null;
			

			const dftDatabase = {
				url: "*LOCAL",
				username: 'xxxx',
				password: 'xxxx'
			}; // url, username, ...

			
			const dftConfig = {}; 
			dftConfig.initialSize = 2;    
			dftConfig.incrementSize =  3; 
			dftConfig.maxSize = 12;        
			dftConfig.idleTimeout = 1_800_000;   
			dftConfig.enableNumericTypeConversion = true;
			dftConfig.debug = true;	
			
			const curLib = "ERPZPRG";
			const libList = "ERPZPRG TEUR4 EURO4 QTEMP QGPL DEBUG LDPRX PROEDI_FIC PROEDI_INT PROEDI_400 JPLTOOLS LIBHTTP ERPZSRC PEEK ERPZDCT";

			try {	
				// Configuration
				pool = new idbPConnector.DBPool(dftDatabase, dftConfig);
				poolConnection = pool.attach();
				
				await poolConnection.connection.setConnAttr(SQL_ATTR_DBC_SYS_NAMING, SQL_TRUE);
				
				//let _listLIBL = libList.split(' ');
				//await poolConnection.connection.setLibraryList(_listLIBL);
				
				let statement = await poolConnection.newStatement();
				
				// LibList + curLib
				const sqlChgLibl = `CALL QSYS2.QCMDEXC('CHGLIBL LIBL(${libList}) CURLIB(${curLib})')`;
				await statement.exec(sqlChgLibl);
				//await statement.close();
				
				// Test sql
				//statement = await poolConnection.newStatement();
			
				let sql = `SELECT rtrim(F_PRIMA('01V4')) as CAL0005 FROM FACENT A  LIMIT 50 WITH UR  `;
			
				let resultSet =  await statement.exec(sql);	
				await statement.close();
				
				let nbRows = resultSet?.length || 0;
				console.log(` execSql ok: ${nbRows} rows `);
		 
			}
			catch(err){
				let msgerr = `test error: ${err.stack}`;
				console.log(msgerr, err);
			}
			finally{
				poolConnection.detach();
			}
		}

		//MAIN...
		testsql();

Debug traces:

	Start...
	Creating Connection 0...
	Connection 0 created
	Creating Connection 1...
	Connection 1 created
	Creating Connection 2...
	Connection 2 created
	Finding available Connection...
	Connection 0 found
	SetConnAttr() attr = 10004, value = 1, rc = 0
	SQLFreeStmt: stmth 3 [SQL_DROP]
	SQLFreeStmt(0)
	SQLExecDirect(0): CALL QSYS2.QCMDEXC('CHGLIBL LIBL(ERPZPRG TEUR4 EURO4 QTEMP QGPL DEBUG LDPRX PROEDI_FIC PROEDI_INT PROEDI_400 JPLTOOLS LIBHTTP ERPZSRC PEEK ERPZDCT) CURLIB(ERPZPRG)')
	SQLNUMRESULTSCOLS(0) Column Count = 0
	NO RESULTS: SQLExecDirect() callSQLExecDirect(-1): SELECT rtrim(F_PRIMA('01V4')) as CAL0005
	FROM FACENT A 
	 LIMIT 50 WITH UR 
		
	test error: Error: SQLSTATE=42704 SQLCODE=-204 F_PRIMA of type *N dans *LIBL not found.
		at Object.<anonymous> (/QOpenSys/QIBM/TestsOD/testPConnector/node_modules/idb-pconnector/lib/statement.js:175:18) Error: SQLSTATE=42704 SQLCODE=-204 F_PRIMA of type *N dans *LIBL not found.
		at Object.<anonymous> (/QOpenSys/QIBM/TestsOD/testPConnector/node_modules/idb-pconnector/lib/statement.js:175:18)
	SQLFreeStmt: stmth 3 [SQL_DROP]
	SQLFreeStmt(0)

Is this a problem with my code or is the idb-pconnector blocking it?
Thank you in advance for any help or suggestions.

Didier

**Comment by abmusse** [comment]
From a statement you should use execute(): https://github.com/IBM/nodejs-idb-pconnector/tree/master/docs#statementexecute

exec() does not work with stored procedures

```
Statement.exec(sql)
Directly executes a given sql String. The exec() method does not work with stored procedure use execute() method instead.
```

You can also use https://github.com/IBM/nodejs-idb-pconnector/tree/master/docs#dbpoolprepareexecutesql-params-options for the pool
